mininet/link.py

```diff
@@ -118,14 +143,34 @@ def updateAddr( self ):
         "Return IP address and MAC address based on ifconfig."
         ifconfig = self.ifconfig()
         ips = self._ipMatchRegex.findall( ifconfig )
+        ip6s = self._ipMatchRegex.findall( ifconfig )
         macs = self._macMatchRegex.findall( ifconfig )
         self.ip = ips[ 0 ] if ips else None
+        self.ip6 = None
+        self.ip6ll = None
+        for addr, scope in ip6s:
+            if scope == 'link':
+                if self.ip6ll is None:
+                    self.ip6ll = addr
+            else:
+                if self.ip6 is None:
+                    self.ip6 = addr
         self.mac = macs[ 0 ] if macs else None
-        return self.ip, self.mac
+        return self.IP(), self.mac
+
+    def IP6LinkLocal( self ):
+        "Return link-local IP address"
+        return self.ip6ll
+
+    def IP6( self ):
+        "Return IPv6 address"
+        return self.ip6
 
     def IP( self ):
         "Return IP address"
-        return self.ip
+        if self.ip is not None:
+            return self.ip
+        return self.ip6
```

Comment:
Return IPv6 address if an IPv4 address is not available.  Mostly backwards compatible.